lib/ChildUtils.spec.lua

```diff
@@ -0,0 +1,84 @@
+return function()
+	local ChildUtils = require(script.Parent.ChildUtils)
+	local createElement = require(script.Parent.createElement)
```

Comment:
Can we get away with using dummy tables as children instead of elements?

I think I updated `iterateChildren` to no longer care about the type of the children since this function would also be useful for iterating over the children of nodes if we go that route.

Reply:
The blocker to that when I was initially writing those tests was that we use the rich type in `iterateChildren` to determine if we were passed an element, or a table of elements - they're both tables in the eyes of `typeof`:

https://github.com/Roblox/roact/blob/ef681f3ce2a45dd68f96d38023daf3051e6e2125/lib/ChildUtils.lua#L39

We could just falsify the type, but that does feel a little weird.

Reply:
Hm, I guess that's true, this seems fine then.